rclpy/rclpy/serialization.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
+from rclpy.type_support import check_for_type_support
+
+
+def deserialize_message(serialized_message: bytes, message_type):
+    """
+    Deserialize a ROS message.
+
+    :param serialized_message: The ROS message to deserialized.
+    :param message_types: The type of the serialized ROS message.
+    :return: The deserialized ROS message.
+    """
+    # this line imports the typesupport for the message module if not already done
+    check_for_type_support(message_type)
+    return _rclpy.rclpy_deserialize(serialized_message, message_type)
+
+
+def serialize_message(message) -> bytes:
+    """
+    Serialize a ROS message.
+
+    :param message: The ROS message to serialize.
+    :return: The serialized bytes.
+    """
+    # this line imports the typesupport for the message module if not already done
+    check_for_type_support(message)
```

Comment:
According to the docblock of `check_for_type_support` the function expects a message type - not a message?

Reply:
I guess? It looks like both the message and the type have the same attributes that it expects, e.g.
  
        msg = BasicTypes()
        msg.__class__._TYPE_SUPPORT
        msg.__class__.__import_type_support__()
        BasicTypes.__class__._TYPE_SUPPORT
        BasicTypes.__class__.__import_type_support__()

I'm abusing this in the C-extension too:

https://github.com/ros2/rclpy/blob/668c9b9e3ff07ac78281d80a15394a38f896860a/rclpy/src/rclpy/_rclpy.c#L4703-L4704

To err on the side of caution, I've switched to using types ac346e4

Reply:
I see. Maybe updating the docblock of that function would be good to state that it can be used with a message instance too. (Not necessary as part of this PR.)